plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/validate-state.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+import { dispatch, select } from '@wordpress/data';
+import { ShippingAddress } from '@woocommerce/settings';
+import { VALIDATION_STORE_KEY } from '@woocommerce/block-data';
+import { __ } from '@wordpress/i18n';
+
+function previousString( initialValue?: string ) {
+	let lastValue = initialValue;
+
+	function track( value: string ) {
+		const currentValue = lastValue;
+
+		lastValue = value;
+
+		// Return the previous value
+		return currentValue;
+	}
+
+	return track;
+}
+
+const lastValue = previousString( '' );
+
+export const validateState = (
+	addressType: string,
+	values: ShippingAddress,
+	isRequired: boolean
+) => {
+	const validationErrorId = `${ addressType }_state`;
+	const hasValidationError =
+		select( VALIDATION_STORE_KEY ).getValidationError( validationErrorId );
+
+	const countryChanged = lastValue( values.country ) !== values.country;
+
+	if ( hasValidationError ) {
+		if ( ! isRequired || values.state ) {
+			// Validation error has been set, but its no longer required, or the state was provided, clear the error.
+			dispatch( VALIDATION_STORE_KEY ).clearValidationError(
+				validationErrorId
+			);
+		} else if ( ! countryChanged ) {
+			// Validation error has been set, there has not been a country set so show the error.
+			dispatch( VALIDATION_STORE_KEY ).showValidationError(
+				validationErrorId
+			);
+		}
+	} else if (
+		! hasValidationError &&
+		isRequired &&
+		! values.state &&
+		values.country
+	) {
+		// No validation has been set yet, if it's required, there is a country set and no state, set the error.
+		dispatch( VALIDATION_STORE_KEY ).setValidationErrors( {
+			[ validationErrorId ]: {
+				message: __( 'Please select your state', 'woocommerce' ),
```

Comment:
error should `Please select a ${stateField.label}`

Reply:
@senadir should it be the same for the country validation too then?

https://github.com/woocommerce/woocommerce/blob/9a367642a29a1d5b7f3c74f39469402bf4fa2ea6/plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/validate-country.ts#L28-L34

Reply:
I've fixed the state issue, I don't mind fixing the country here too, but could also be done as a follow up since this behaviour was already there.